### PR TITLE
meta: automatically update PRs that have auto merge enabled

### PR DESCRIPTION
**Pull request by ephys**
This PR adds a replacement for the merge queue. Instead of using the merge queue, we can assign the `automerge` label to a PR to make the CI automatically merge the PR when ready, as well as update it if it becomes outdated.

Added the `automerge` label on this PR to test it.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -16,12 +16,13 @@ If unsure about something, just do as best as you're able.
...
-## Todos
+## List of Breaking Changes
 
-- [ ] <!-- e.g. #1 feature: Extend the type script definition -->
-- [ ] <!-- e.g. #2 test: Does this also work with MySQL 8? -->
-- [ ] <!-- ... -->
+<!-- If you have any breaking changes, you should list them below. -->
+<!-- Anything added below this comment group will automatically be added to our changelog. -->
+<!-- Do not modify the next line! It is used by our bot! -->
+<!-- [BREAKING CHANGES START] -->
```
Comment (WikiRik) [inline .github/PULL_REQUEST_TEMPLATE.md:28]: This is nice!

File: .github/workflows/automerge.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Automerge PR
+on:
+  pull_request:
+    types:
```
Comment (WikiRik) [inline .github/workflows/automerge.yml:4]: Do we need all these types?

Comment (WikiRik) [reply .github/workflows/automerge.yml:4]: Nevermind, you removed a few. This should be fine then

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (ephys) [reply .github/PULL_REQUEST_TEMPLATE.md]: I've tweaked it so the marker can be visible as we need to make sure a blank space is present between the marker and the first breaking change, if we want to follow the conventions of our changelog generator

File: .github/workflows/automerge.yml (hunk 2 above)
Comment (ephys) [reply .github/workflows/automerge.yml:4]: Please ignore - testing that replying does not trigger Automerge

Comment (ephys) [reply .github/workflows/automerge.yml:4]: Yay it worked: https://github.com/sequelize/sequelize/actions/runs/7944286727

**Comment by ephys** [comment]
Hmmm. The CI checks are not running and I don't know if that's because the PR was a draft or if it's because a bot added the merge commit

**Comment by ephys** [comment]
OK! This new version does not merge in the target branch anymore. Instead, we enable github's native auto-merge, and this new action will run every time a PR is added on `main`. It will merge `main` back into every PR that targets it, if they have github's auto merge enabled

**Review by WikiRik — APPROVED** [review]

**Pull request merged**